src/wormhole_transit_relay/transit_server.py

```diff
@@ -283,14 +286,14 @@ def connection_got_token(self, token, new_side, new_tc):
 
                 # drop and stop tracking the rest
                 potentials.remove(old)
-                for (_, leftover_tc) in potentials:
+                for (_, leftover_tc) in potentials.copy():
                     # Don't record this as errory. It's just a spare connection
                     # from the same side as a connection that got used. This
                     # can happen if the connection hint contains multiple
                     # addresses (we don't currently support those, but it'd
                     # probably be useful in the future).
                     leftover_tc.disconnect_redundant()
-                self._pending_requests.pop(token)
+                self._pending_requests.pop(token, None)
```

Comment:
`pop` will fail if `token` is no longer present in the set (as occurs in testing, see above comment).

Adding `None` as the default return value allows us to continue if `token not in self._pending_requests`